source/modulo_components/modulo_components/component_interface.py

```diff
@@ -0,0 +1,121 @@
+from rclpy.node import Node
+from tf2_ros.buffer import Buffer
+from tf2_ros.transform_listener import TransformListener
+from tf2_ros import TransformBroadcaster
+from rcl_interfaces.msg import ParameterDescriptor
+from std_msgs.msg import Bool
+
+from typing import Union
+from typing import Callable
+
+
+class ComponentInterface(Node):
+    """
+    Abstract class to represent a Component in python, following the same logic pattern
+    as the c++ modulo_component::ComponentInterface class.
+    ...
+    Attributes:
+        predicates (dict(Parameters(bool))): map of predicates added to the Component.
+        predicate_publishers (dict(rclpy.Publisher(Bool))): map of publishers associated to each predicate.
+        tf_buffer (tf2_ros.Buffer): the buffer to lookup transforms published on tf2.
+        tf_listener (tf2_ros.TransformListener): the listener to lookup transforms published on tf2.
+        tf_broadcaster (tf2_ros.TransformBroadcaster): the broadcaster to publish transforms on tf2
+    Parameters:
+        period (double): period (in s) between step function calls.
+        has_tf_listener (bool): if true, add a tf listener to enable calls to lookup transforms.
+        has_tf_broadcaster (bool): if true, add a tf broadcaster to enable the publishing of transforms.
+    """
+
+    def __init__(self, node_name, *kargs, **kwargs):
+        """
+        Constructs all the necessary attributes and declare all the parameters.
+            Parameters:
+                node_name (str): name of the node to be passed to the base Node class
+        """
+        super().__init__(node_name, *kargs, **kwargs)
+        self.declare_parameter('period', 0.1,
+                               ParameterDescriptor(description="Period (in s) between step function calls."))
+        self.declare_parameter('has_tf_listener', False,
+                               ParameterDescriptor(
+                                   description="If true, add a tf listener to enable calls to lookup transforms."))
+        self.declare_parameter('has_tf_broadcaster', False,
+                               ParameterDescriptor(
+                                   description="If true, add a tf broadcaster to enable the publishing of transforms."))
+
+        self._period = self.get_parameter('period').get_parameter_value().double_value
+        self._has_tf_listener = self.get_parameter('has_tf_listener').get_parameter_value().double_value
+        self._has_tf_broadcaster = self.get_parameter('has_tf_broadcaster').get_parameter_value().double_value
+        self._predicates = {}
+        self._predicates_publishers = {}
```

Comment:
I would prefer to keep the general rule of "single, plural" for collections rather than "plural, plural". The "pencil sharpener" vs. "pencils sharpeners" example comes to mind.
```suggestion
        self._predicate_publishers = {}
```

Reply:
This has been updated in all occurences